File: cpp/Program/Motorcycle.cpp
#pragma once
#include <iostream>
#include <string>
#include "Vehicle.cpp"

using namespace std;

class Motorcycle : public Vehicle
{
private:
    string jenisMotor;
    int kapasitasTangki;


public:
    Motorcycle(){

    }

    Motorcycle(string jenisMotor, string platNomor, string merk, string tahunProduksi, string warna) : Vehicle(platNomor, merk, tahunProduksi, warna){
        this->jenisMotor = jenisMotor;
        kapasitasTangki = 0;
    }

    Motorcycle(string jenisMotor, int kapasitasTangki, string platNomor, string merk, string tahunProduksi, string warna) : Vehicle(platNomor, merk, tahunProduksi, warna){
        this->jenisMotor = jenisMotor;
        this->kapasitasTangki = kapasitasTangki;
    }

    string GetJenisMotor()  {
        return jenisMotor;
    }

    void SetJenisMotor(string jenisMotor) {
        this->jenisMotor = jenisMotor;
    }

    int GetKapasitasTangki()  {
        return kapasitasTangki;
    }

    void SetKapasitasTangki(int kapasitasTangki) {
        this->kapasitasTangki = kapasitasTangki;
    }

    ~Motorcycle(){

    }
};

File: python/Program/Motorcycle.py
from Vehicle import Vehicle

class Motorcycle(Vehicle):
    __jenisMotor = ""
    __kapasitasTangki = ''
    
    def __init__(self,platNomor = "", merk = "", tahunProduksi = "", warna = "", jenisMotor = "", kapasitasTangki = ''):
        super().__init__(platNomor, merk, tahunProduksi, warna)
        self.__jenisMotor = jenisMotor
        self.__kapasitasTangki = kapasitasTangki
    

    def get_jenisMotor(self):
        return self.__jenisMotor

    def set_jenisMotor(self, value):
        self.__jenisMotor = value

    def get_kapasitasTangki(self):
        return self.__kapasitasTangki

    def set_kapasitasTangki(self, value):
        self.__kapasitasTangki = value

    